M2/Macaulay2/packages/Varieties.m2

```diff
@@ -382,40 +482,63 @@ degreeList := M -> (
 
 -- quotienting by local H_m^0(M) to "saturate" M
 -- TODO: use irrelevant ideal here
-killH0 := M -> if (H0 := saturate(0*M)) == 0 then M else M / H0
+killH0 := -*(cacheValue symbol TorsionFree)*- (M -> if (H0 := saturate(0*M)) == 0 then M else M / H0)
 
 -- TODO: add tests:
 -- - global sections of sheafHom are Hom
--- TODO: implement for multigraded ring
+-- TODO: implement for multigraded ring using emsbound
 -- TODO: this can change F.module to the result!
+-- NOTE: this may have elements in degrees below bound as well, is that a bug?
 twistedGlobalSectionsModule = (F, bound) -> (
     -- compute global sections module Gamma_(d >= bound)(X, F(d))
     A := ring F;
+    -- FIXME: this line, as opposed to
+    --  cokernel presentation module F
+    -- breaks the test added in 975d780470.
+    -- However, we need to keep the information
+    -- cached in M, for instance if M is a Hom module.
+    M := module F;
     if degreeLength A =!= 1 then error "expected degree length 1";
-    -- quotient by H_m^0(M)
-    M := killH0 cokernel presentation module F;
+    -- quotient by HH^0_m(M) to kill the torsion
+    -- TODO: pass the appropriate irrelevant ideal
+    N := killH0 M;
```

Comment:
Why is the second ```killH0``` required?

Reply:
There is just one. The red one was removed and the green one was added.